sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala

```diff
@@ -407,6 +407,39 @@ class FileSourceStrategySuite extends QueryTest with SharedSQLContext with Predi
     }
   }
 
+  test("[SPARK-16818] partition pruned file scans implement sameResult correctly") {
+    withTempPath { path =>
+      val tempDir = path.getCanonicalPath
+      spark.range(100)
+        .selectExpr("id", "id as b")
+        .write
+        .partitionBy("id")
+        .parquet(tempDir)
+      val df = spark.read.parquet(tempDir)
+      def getPlan(df: DataFrame): SparkPlan = {
+        df.queryExecution.executedPlan
+      }
+      assert(getPlan(df.where("id = 2")).sameResult(getPlan(df.where("id = 2"))))
+      assert(!getPlan(df.where("id = 2")).sameResult(getPlan(df.where("id = 3"))))
+    }
+  }
+
+  test("[SPARK-16818] exchange reuse respects differences in partition pruning") {
+    spark.conf.set("spark.sql.exchange.reuse", true)
```

Comment:
Oh, I assumed the test already did so since I've seen this pattern
elsewhere. If it affects more than just the suite, I can submit a follow-up
fix.

On Sat, Jul 30, 2016, 10:59 PM Reynold Xin <notifications@github.com> wrote:

> In
> sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala
> <https://github.com/apache/spark/pull/14427#discussion_r72898521>:
>
> > +      spark.range(100)
> > +        .selectExpr("id", "id as b")
> > +        .write
> > +        .partitionBy("id")
> > +        .parquet(tempDir)
> > +      val df = spark.read.parquet(tempDir)
> > +      def getPlan(df: DataFrame): SparkPlan = {
> > +        df.queryExecution.executedPlan
> > +      }
> > +      assert(getPlan(df.where("id = 2")).sameResult(getPlan(df.where("id = 2"))))
> > +      assert(!getPlan(df.where("id = 2")).sameResult(getPlan(df.where("id = 3"))))
> > +    }
> > +  }
> > +
> > +  test("[SPARK-16818] exchange reuse respects differences in partition pruning") {
> > +    spark.conf.set("spark.sql.exchange.reuse", true)
>
> ah actually just realized we could've improved with by using "withSQLConf"
> -- it makes sure the configs get reset after the test case finishes running.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/spark/pull/14427/files/ef60367331fb3097040cfb0849bdc314c8d399ea#r72898521>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAA6Sun3Ai6lEtCs9dcjaxlAtO3Y_a2Qks5qbDnQgaJpZM4JY91Z>
> .
>